src/main/java/uk/gov/pay/ledger/filters/LoggingMDCRequestFilter.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.pay.ledger.filters;
+
+import org.slf4j.MDC;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import java.io.IOException;
+import java.util.Optional;
+
+import static uk.gov.pay.logging.LoggingKeys.GATEWAY_ACCOUNT_ID;
+import static uk.gov.pay.logging.LoggingKeys.LEDGER_EVENT_ID;
+
+public class LoggingMDCRequestFilter implements ContainerRequestFilter {
+
+    public static final String PARENT_EXTERNAL_ID = "parent_external_id";
+    public static final String TRANSACTION_EXTERNAL_ID = "transactionExternalId";
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        getPathParameterFromRequest("accountId", requestContext)
+                .ifPresent(accountId -> MDC.put(GATEWAY_ACCOUNT_ID, accountId));
+
+        getPathParameterFromRequest("eventId", requestContext)
+                .ifPresent(eventId -> MDC.put(LEDGER_EVENT_ID, eventId));
+
+        getPathParameterFromRequest("parent_external_id", requestContext)
```

Comment:
this should be "parentExternalId" as per https://github.com/alphagov/pay-ledger/blob/master/src/main/java/uk/gov/pay/ledger/transaction/resource/TransactionResource.java#L106

Reply:
Done